src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeEvent.java

```diff
@@ -97,14 +108,25 @@ private GenericRecord asIcebergRecord(Types.StructType tableFields, JsonNode dat
     return record;
   }
 
-  private Type.PrimitiveType icebergFieldType(String fieldType) {
+  private Type.PrimitiveType icebergFieldType(String fieldType, String fieldTypeName) {
     switch (fieldType) {
       case "int8":
       case "int16":
       case "int32": // int 4 bytes
-        return Types.IntegerType.get();
+        if (IcebergChangeEvent.coerceDebeziumDate && fieldTypeName.equals("io.debezium.time.Date")) {
+          return Types.StringType.get();
```

Comment:
Date primitives are supported in Iceberg -- any reason we wouldn't use that here?